src/coreclr/nativeaot/System.Private.Reflection.Execution/src/Internal/Reflection/Execution/ExecutionEnvironmentImplementation.ManifestResources.cs

```diff
@@ -78,74 +73,35 @@ private static unsafe UnmanagedMemoryStream ReadResourceFromBlob(ResourceInfo re
             return new UnmanagedMemoryStream(pBlob + resourceInfo.Index, resourceInfo.Length);
         }
 
-        private static LowLevelList<ResourceInfo> GetExtractedResources(Assembly assembly)
+        private static IEnumerable<ResourceInfo> ExtractResources(Assembly assembly)
         {
-            LowLevelDictionary<string, LowLevelList<ResourceInfo>> extractedResourceDictionary = ExtractedResourceDictionary;
+            ArgumentNullException.ThrowIfNull(assembly);
```

Comment:
How can we end up with null assembly here?

Argument validation on internal methods is typically done by asserts only. We typically throw argument exceptions for public surface only.

Reply:
It's always called with `this` as assembly, so there won't be null.